runtime/compilers/rillv1/parser.go

```diff
@@ -411,18 +431,17 @@ func (p *Parser) reparseExceptRillYAML(ctx context.Context, paths []string) (*Di
 		// Watcher sends multiple events for a single edit. We want to only restart the controller when .env actually changes.
 		// So we check the new .env contents against the contents stored in parser state.
 		if isDotEnv {
-			oldDotEnv := p.DotEnv
-			err := p.parseDotEnv(ctx, checkPaths[i])
+			oldDotEnv := p.DotEnv[path]
+			err := p.parseDotEnv(ctx, path)
 			if err == nil {
-				modifiedDotEnv = !maps.Equal(p.DotEnv, oldDotEnv)
+				modifiedDotEnv = !maps.Equal(p.DotEnv[path], oldDotEnv)
```

Comment:
This looks right now, I think. Given my comment above, I understand it's confusing to see a `parseDotEnv` call here (since that should only happen in the second phase, not the "cleanup" phase here). The reason it calls it here is a hacky way to easily work out if the `.env` file has been modified.